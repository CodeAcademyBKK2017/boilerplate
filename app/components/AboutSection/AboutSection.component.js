import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './AboutSection.style';
import Touchable from 'react-native-platform-touchable';
import {
  Text,
  View
} from 'react-native';

export default class AboutSection extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Touchable
          style={styles.aboutButtonTouch}
          background={Touchable.Ripple('orange')}
          onPress={this.props.onAboutButtonPress}
        >
          <Text style={styles.aboutButtonContent}>About</Text>
        </Touchable>
      </View>
    );
  }
}

AboutSection.propTypes = {
  onAboutButtonPress: PropTypes.func.isRequired
};

AboutSection.defaultProps = {
  onAboutButtonPress: noop
};
