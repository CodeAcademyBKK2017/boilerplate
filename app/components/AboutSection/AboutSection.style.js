import {
  StyleSheet
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    alignItems: 'stretch'
  },
  aboutButtonTouch: {
    backgroundColor: '#ddd'
  },
  aboutButtonContent: {
    paddingVertical: 10,
    paddingHorizontal: 10,
    textAlign: 'center'
  }
});

export default styles;
