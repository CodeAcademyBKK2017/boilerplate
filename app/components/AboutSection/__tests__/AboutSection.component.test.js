import 'react-native';
import AboutSection from '../AboutSection.component';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('AboutSection', () => {
  it('renders correctly', () => {
    const snapshot = renderer.create(<AboutSection/>).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
