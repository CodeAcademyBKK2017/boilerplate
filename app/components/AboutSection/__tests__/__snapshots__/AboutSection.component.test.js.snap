// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutSection renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderTopWidth": 0.5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 10,
          "textAlign": "center",
        }
      }
    >
      About
    </Text>
  </View>
</View>
`;
